From 177a5fc5721b82cf680f2c7c2cc61bd0dc78b701 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Wed, 19 Jun 2019 16:28:10 -0400
Subject: [PATCH 21/35] Lock down /proc/kcore

Message-id: <20190619162824.427-6-labbott@redhat.com>
Patchwork-id: 265695
O-Subject: [ARK PATCH 21/35] Lock down /proc/kcore
Bugzilla:
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

From: David Howells <dhowells@redhat.com>

Disallow access to /proc/kcore when the kernel is locked down to prevent
access to cryptographic data.

Signed-off-by: David Howells <dhowells@redhat.com>
Reviewed-by: James Morris <james.l.morris@oracle.com>
Signed-off-by: Matthew Garrett <matthewgarrett@google.com>
---
 fs/proc/kcore.c | 2 ++
 1 file changed, 2 insertions(+)
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 fs/proc/kcore.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/proc/kcore.c b/fs/proc/kcore.c
index f5834488b67d..0639228c4904 100644
--- a/fs/proc/kcore.c
+++ b/fs/proc/kcore.c
@@ -545,6 +545,8 @@ read_kcore(struct file *file, char __user *buffer, size_t buflen, loff_t *fpos)
 
 static int open_kcore(struct inode *inode, struct file *filp)
 {
+	if (kernel_is_locked_down("/proc/kcore"))
+		return -EPERM;
 	if (!capable(CAP_SYS_RAWIO))
 		return -EPERM;
 
-- 
2.21.0

